File: app/src/main/java/io/github/pau1adam/kotlinjavasamples/java/Food2ForkApi.java
package io.github.pau1adam.kotlinjavasamples.java;

import io.github.pau1adam.kotlinjavasamples.kotlin.GetRecipeResponse;
import io.github.pau1adam.kotlinjavasamples.kotlin.GetSearchRecipeResponse;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface Food2ForkApi {
    @GET("search")
    Call<GetSearchRecipeResponse> search(
            @Query("key") String key,
            @Query("q") String keyword,
            @Query("page") int page
    );

    @GET("get")
    Call<GetRecipeResponse> getRecipe(
            @Query("key") String key,
            @Query("rId") String recipeId
    );
}

File: app/src/main/java/io/github/pau1adam/kotlinjavasamples/Food2ForkApi.kt
package io.github.pau1adam.kotlinjavasamples

import kotlinx.coroutines.experimental.Deferred
import retrofit2.http.GET
import retrofit2.http.Query


interface Food2ForkApi {

    @GET("search")
    fun search(
            @Query("key") key: String,
            @Query("q") keyword: String,
            @Query("page") page: Int?
    ): Deferred<GetSearchRecipeResponse>

    @GET("get")
    fun getRecipe(
            @Query("key") key: String,
            @Query("rId") recipeId: String
    ): Deferred<GetRecipeResponse>
}